toml.abnf

```diff
@@ -1,6 +1,10 @@
 ;; This document describes TOML's syntax, using the ABNF format (defined in
 ;; RFC 5234 -- https://www.ietf.org/rfc/rfc5234.txt).
 ;;
+;; Although a TOML document must be valid UTF-8, this grammar is written to
+;; apply to the string of non-negative integers that are the Unicode Scalar
+;; Values obtained from decoding the UTF-8 input.
```

Comment:
Maybe I'm being stupid, but I don't really follow what "non-negative integers that are the Unicode Scalar Values obtained from decoding the UTF-8 input" means.

I *think* you mean to say that this document uses Unicode codepoints, rather than UTF-8 byte-encoding?

Reply:
> Maybe I'm being stupid, but I don't really follow what "non-negative integers that are the Unicode Scalar Values obtained from decoding the UTF-8 input" means.

You're not stupid; I came here to make the same observation. No idea what that sentence is trying to convey. Seems like it's supposed to be a disambiguation of sorts? Has the opposite effect O_o

Reply:
Thanks for the feedback.  It is important to be understandable.  I simplified the language (commit [1f8a780](https://github.com/toml-lang/toml/pull/876/commits/1f8a780805e136e1e10e8abbf272dd245e2843f4)).  
The reason I think a paragraph like this will be helpful to some people is because the spec `toml.md` states "A TOML file must be a valid UTF-8 encoded Unicode document", but the grammar does not refer to UTF-8 "code units" (byte values).  There is a decoding step prior to parsing.  Hopefully the new language is acceptable.

The most important part of this PR, though, is to fix issue #873 by adding CRs on checkout.

Reply:
@bendyarm Ah, I see. Thanks for the clarification.